CMakeLists.txt

```diff
@@ -1,47 +1,66 @@
 cmake_minimum_required(VERSION 3.3)
-project(qmarkdowntextedit)
+project(qmarkdowntextedit LANGUAGES CXX)
 
 #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+#set(CMAKE_CXX_STANDARD 11) Use this instead
+#set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTOUIC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-find_package( Qt5Core REQUIRED )
-find_package( Qt5Widgets REQUIRED )
-find_package( Qt5Gui REQUIRED )
+# add option to disable test executable
+option(QMARKDOWNTEXTEDIT_EXE "Build test executable" ON)
 
-qt5_wrap_ui(ui_qplaintexteditsearchwidget.h qplaintexteditsearchwidget.ui)
+# find qt
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Svg)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Svg)
 
-set(RESOURCE_FILES
-        media.qrc
-        )
+# needed for windows
+if(WIN32)
+    set(INTL_LDFLAGS -lintl)
+endif(WIN32)
 
-qt5_add_resources(RESOURCE_ADDED ${RESOURCE_FILES})
+# QMarkdownTextEdit library
+set(RC_FILES
+    media.qrc
+)
 
-set(SOURCE_FILES
+# Translations arent loaded so don't include them
+set(TS_FILES
+    trans/qmarkdowntextedit_de.ts
+    trans/qmarkdowntextedit_ur.ts
+    trans/qmarkdowntextedit_zh_CN.ts
+)
+
+set(QMARKDOWNTEXTEDIT_FILES
+    markdownhighlighter.h
     markdownhighlighter.cpp
     linenumberarea.h
-    main.cpp
-    mainwindow.cpp
-    mainwindow.ui
+    qmarkdowntextedit.h
     qmarkdowntextedit.cpp
+    qownlanguagedata.h
     qownlanguagedata.cpp
+    qplaintexteditsearchwidget.h
     qplaintexteditsearchwidget.ui
     qplaintexteditsearchwidget.cpp
-    )
+    media.qrc
+)
 
-add_executable(qmarkdowntextedit ${SOURCE_FILES} ${RESOURCE_ADDED})
+add_library(qmarkdowntextedit ${QMARKDOWNTEXTEDIT_FILES})
 
-include_directories(${Qt5Widgets_INCLUDES})
+target_link_libraries(qmarkdowntextedit Qt${QT_VERSION_MAJOR}::Widgets Qt${QT_VERSION_MAJOR}::Svg ${INTL_LDFLAGS})
```

Comment:
Why link with Qt svg?

Reply:
Because QPlainTextEditSearchWidget loads Svg images

Reply:
Does loading svg image means we have to explicitly link to qt svg? I would think not. Qt svg would be required if we used qt svg apis

Reply:
I tested it and I was wrong. I have now removed it.